src/computation_cache.h

```diff
@@ -168,10 +169,23 @@ class InfoScoreCache {
     score_map_.insert({ScoreKey(X, Y, Z, ui), score});
   }
 
+  pair<double, bool> getEntropy(int X, int Y, int Z) {
+    auto it = entropy_map_.find(ScoreKey(X, Y, Z, vector<int>()));
```

Comment:
Maybe we could add a check on ScoreKey(Y, X, Z) here too to make (X, Y) symmetric? But I doubt we will ever need this in practice, since the order of pairs is fixed. So I guess it's fine here

Reply:
Isn't it symmetric already ? (X,Y) are in a set in ScoreKey

```
struct ScoreKey {
  set<int> XY;
  int Z;
  set<int> ui;

  ScoreKey(int X, int Y, int Z, const vector<int>& ui)
      : XY({X, Y}), Z(Z), ui(begin(ui), end(ui)) {}

  bool operator<(const ScoreKey& other) const {
    if (XY == other.XY) {
      if (Z == other.Z) {
        return ui < other.ui;
      }
      return Z < other.Z;
    }
    return XY < other.XY;
  }
};
```

Reply:
You are right, I forgot what I wrote :)

Reply:
Also it didn't feel amazing to re-use this key but at least it has something to do with the score computation

Reply:
Yes, that was exactly my thoughts when looking at the empty vector in `ScoreKey(X, Y, Z, vector<int>())`, but another key with only pair(X, Y) and Z feels kind of redundant.